### PR TITLE
Implement _log_softmax | feat(torchlib)

### PR DESCRIPTION
**Pull request by justinchuby**
Fixes https://github.com/microsoft/onnxscript/issues/1077

### CODE REVIEW

**Review by BowenBao — COMMENTED** [review]
Could you also add test coverage?

**Comment by justinchuby** [comment]
> Could you also add test coverage?

On it

**Comment by justinchuby** [comment]
Added tests

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 1]
```diff
@@ -70,6 +70,34 @@ def aten__local_scalar_dense_int(self: IntType) -> INT64:
...
+    # trace_only because we need to cast conditionally based on half_to_float
+    if half_to_float:
+        self = op.Cast(self, to=FLOAT.dtype)
+
+    return aten__log_softmax(self, dim, half_to_float)
+
+
+@torch_op("aten::_log_softmax")
+def aten__log_softmax(
+    self: TFloatHighPrecision, dim: int, half_to_float: bool  # pylint: disable=unused-argument
```
Comment (BowenBao) [inline onnxscript/function_libs/torch_lib/ops/core.py:88]: nit: very minor, should we do below to be consistent (saw in `aten__softmax_half` below) ?
```
    del half_to_float  # Unused
```

File: onnxscript/tests/function_libs/torch_lib/extra_opinfo.py [hunk 2]
```diff
@@ -902,6 +902,34 @@ def sample_inputs_slice_scatter(op_info, device, dtype, requires_grad, **kwargs)
...
+        ((S,), (0,)),
+        ((S, S), (0,)),
+        ((S, S), (1,)),
+        ((S, S), (-1,)),
+        ((S, M, S), (2,)),
+        ((S, 0, 0), (-1,)),
+    ]
+
+    for (shape, dim), half_to_float in itertools.product(cases, (False,)):
+        # NOTE: softmax with half to float conversion is not supported on CPU
```
Comment (BowenBao) [inline onnxscript/tests/function_libs/torch_lib/extra_opinfo.py:927]: Could we test bfloat16?

**Review by BowenBao — APPROVED** [review]

File: onnxscript/tests/function_libs/torch_lib/extra_opinfo.py (hunk 2 above)
Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_lib/extra_opinfo.py:927]: ORT doesn't run bfloat16 on cpu so

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py:88]: This function is not trace only so we couldn't use `del` (not supported by onnxscript yet)

**Pull request merged**